src/practices/Java/JavaDependenciesVersionMajorLevel.ts

```diff
@@ -0,0 +1,73 @@
+import { PracticeContext } from '../../contexts/practice/PracticeContext';
+import { Package } from '../../inspectors/IPackageInspector';
+import { SemverLevel, PackageInspectorBase } from '../../inspectors/package/PackageInspectorBase';
+import { PracticeEvaluationResult, PracticeImpact, ProgrammingLanguage } from '../../model';
+import { DxPractice } from '../DxPracticeDecorator';
+import { IPractice } from '../IPractice';
+import * as axios from 'axios';
+
+@DxPractice({
+  id: 'Java.DependenciesVersionMajorLevel',
+  name: 'Update Dependencies of Major Level',
+  impact: PracticeImpact.medium,
+  suggestion: 'Keep the dependencies updated to have all possible features. Use, for example, Renovate Bot.',
+  reportOnlyOnce: true,
+  url: 'https://renovatebot.com/',
+})
+export class JavaDependenciesVersionMajorLevel implements IPractice {
+  async isApplicable(ctx: PracticeContext): Promise<boolean> {
+    return ctx.projectComponent.language === ProgrammingLanguage.Java || ctx.projectComponent.language === ProgrammingLanguage.Kotlin;
+  }
+
+  async evaluate(ctx: PracticeContext): Promise<PracticeEvaluationResult> {
+    if (ctx.fileInspector === undefined || ctx.packageInspector === undefined) {
+      return PracticeEvaluationResult.unknown;
+    }
+
+    const pkgs = ctx.packageInspector.packages;
+
+    if (pkgs === undefined) {
+      return PracticeEvaluationResult.unknown;
+    }
+
+    const result = await JavaDependenciesVersionMajorLevel.searchMavenCentral(pkgs, 5);
+    const practiceEvaluationResult = JavaDependenciesVersionMajorLevel.isPracticing(result, SemverLevel.major, pkgs);
+
+    return practiceEvaluationResult || PracticeEvaluationResult.practicing;
+  }
+
+  static async searchMavenCentral(pkgs: Package[] | undefined, rows: number) {
+    const latestVersionsJson: { [key: string]: string } = {};
+    const URL = 'http://search.maven.org/solrsearch/select?q=';
+    if (pkgs) {
+      for (const p of pkgs) {
+        const listOfIds = p.name.split(':', 2);
+        const listVersionsEndpoint = `${URL}${listOfIds[0]}+AND+a:${listOfIds[1]}&rows=${rows}&wt=json`;
+        await axios.default.get(listVersionsEndpoint).then((response) => {
+          latestVersionsJson[p.name] = String(response.data.response.docs.pop().latestVersion);
+        });
+      }
+    }
+    return latestVersionsJson;
+  }
+
+  static isPracticing(
+    result: { [key: string]: string },
+    semverVersion: SemverLevel,
+    pkgs: Package[],
+  ): PracticeEvaluationResult | undefined {
+    for (const packageName in result) {
+      const parsedVersion = PackageInspectorBase.semverToPackageVersion(result[packageName]);
+      if (parsedVersion) {
+        for (const pkg of pkgs) {
+          if (pkg.name === packageName) {
+            if (parsedVersion[semverVersion] > pkg.lockfileVersion[semverVersion]) {
+              return PracticeEvaluationResult.notPracticing;
+            }
+          }
+        }
+      }
+    }
+    return undefined;
+  }
+}
```

Comment:
Don't duplicate the code from DependenciesVersionMajorLevel.ts. Make e.g. PracticeUtils.ts instead so there is no duplicated code.

Reply:
Or we can discuss it with @prokopsimek what he thinks about this.

Reply:
> Don't duplicate the code from DependenciesVersionMajorLevel.ts. Make e.g. PracticeUtils.ts instead so there is no duplicated code.

The code in DependenciesVersionMajorLevel.ts will change in @prokopsimek's new PR. I duplicated the code to decouple the classes to make them independent of each other. I agree that we should discuss it with @prokopsimek to see if this is a better design pattern to go for.

Reply:
@ryzzaki see changes in branch `rich-suggestion-text`, this function should be changed to the same logic as there

I agree with something like `src/practices/utils/DependenciesVersionUtils.ts`. :)

Reply:
@prokopsimek I have made `src/practices/utils/DependenciesVersionUtils.ts`, which holds the `packagesToBeUpdated` as a static function. The JS/TS Dependency practices can be easily refactored to use these utils as well.